src/components/LoadingAnimation.js

```diff
@@ -10,29 +13,51 @@ export default function LoadingAnimation({
 }) {
   let _text = text ? text : strings.loadingMsg;
   const [showLoadingScreen, setShowLoadingScreen] = useState(false);
+  const [showReportButton, setShowReportButton] = useState(false);
+  const [showFeedbackModal, setShowFeedbackModal] = useState(false);
 
   useEffect(() => {
     // Code from: https://stackoverflow.com/questions/53090432/react-hooks-right-way-to-clear-timeouts-and-intervals
     // Only show the loading if more that 1000ms have passed.
     let loadingTimer = setTimeout(() => setShowLoadingScreen(true), delay);
+    let showReportButtonTimer = setTimeout(
+      () => setShowReportButton(true),
+      delay + 4000,
+    ); // Wait 4 seconds to show the link to the FeedbackScreen.
     return () => {
       clearTimeout(loadingTimer);
     };
   }, []);
 
   return (
     <>
-      {showLoadingScreen ? (
-        <s.LoadingAnimation style={specificStyle}>
-          <div className="lds-ellipsis">
-            <div></div>
-            <div></div>
-            <div></div>
-            <div></div>
-          </div>
-        </s.LoadingAnimation>
-      ) : (
-        <div></div>
+      <FeedbackModal
+        open={showFeedbackModal}
+        setOpen={() => {
+          setShowFeedbackModal(!showFeedbackModal);
+        }}
+        feedbackOptions={FEEDBACK_OPTIONS.ALL}
+      />
+      {showLoadingScreen && (
+        <s.LoadingContainer>
+          <s.LoadingAnimation style={specificStyle}>
+            <div className="lds-ellipsis">
+              <div></div>
+              <div></div>
+              <div></div>
+              <div></div>
+            </div>
+          </s.LoadingAnimation>
+          {showReportButton && (
```

Comment:
WHOA! Now I think I understand what this is. If the loading animation takes more than 4s we automatically display the Feedback button? This IS POSITIVELY COOL :)

Reply:
It happened already :) 

<img width="1053" alt="image" src="https://github.com/user-attachments/assets/f69860ce-c08e-4158-8135-a0ab41aceec9">